File: 프로그래머스/unrated/120894. 영어가 싫어요/영어가 싫어요.py
def solution(numbers):
    answer = 0
    i=0
    while i < len(numbers):
        answer *= 10
        if numbers[i] == 'z':
            i+=4
        elif numbers[i] == 'o':
            answer += 1
            i+=3
        elif numbers[i] == 't':
            if numbers[i+1] == 'w':
                i+=3
                answer += 2
            else:
                i+=5
                answer += 3
        elif numbers[i] == 'f':
            if numbers[i+1] == 'o':
                i+=4
                answer+=4
            else:
                i+=4
                answer+=5
        elif numbers[i] == 's':
            if numbers[i+1] == 'i':
                i+=3
                answer+=6
            else:
                i+=5
                answer+=7
        elif numbers[i] == 'e':
            i+=5
            answer+=8
        else:
            i+=4
            answer+=9
    return answer

File: 프로그래머스/Lv.0/120894. 영어가 싫어요/영어가 싫어요.java
class Solution {
    public long solution(String numbers) {
        long answer = 0;
        String[] num = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        
        for(int i=0;i<10;i++){
            numbers = numbers.replaceAll(num[i], String.valueOf(i));
        }
        
        return Long.parseLong(numbers);
    }
}